integrations/terraform/envs/portal/read_portal_config.py

```diff
@@ -1,61 +1,192 @@
 import os
+from typing import Any, Dict
+
 import requests
 import yaml
 
 # API key of the data product portal
-API_KEY = os.getenv('API_KEY')
+API_KEY: str = os.getenv("API_KEY", "")
 # URL of where to contact the portal API
 API_HOST = "https://portal.acme.com"
-FOLDER = './'
-proxies = {}
+FOLDER = "./"
+proxies: Dict[str, str] = {}
 
-HEADERS = {
-    "x-key": f"{API_KEY}"
-}
+HEADERS = {"x-key": f"{API_KEY}"}
 session = requests.Session()
 session.proxies = proxies
-session.headers = HEADERS
+session.headers.update(HEADERS)
 
-def get_data_products():
-    result = session.get(f"{API_HOST}/api/data_products")
-    if 'correlation_id' in result.json():
-        print(result.json())
+
+# Definitions to allow for values to be exported as quoted strings in the yaml outputs
+class QuotedString(str):
+    pass
+
+
+def quoted_scalar(dumper: yaml.Dumper, data: Any):
+    return dumper.represent_scalar("tag:yaml.org,2002:str", data, style='"')
+
+
+yaml.Dumper.add_representer(QuotedString, quoted_scalar)
+
+
+# Verification of API responses
+def verify_response(response: Dict[str, Any]):
+    if "correlation_id" in response:
+        print(response)
         exit(1)
-    data_products = {}
-    for data_product_info in result.json():
-        data_product = session.get(f"{API_HOST}/api/data_products/{data_product_info.get('id')}")
-        data_product = data_product.json()
+
+
+def get_data_products():
+    data_products = session.get(f"{API_HOST}/api/data_products").json()
+    verify_response(data_products)
+
+    data_products_export = {}
+    for data_product_info in data_products:
+        data_product = session.get(
+            f"{API_HOST}/api/data_products/{data_product_info.get('id')}"
+        ).json()
+        verify_response(data_product)
+
         datasets = []
-        for dataset in data_product.get("dataset_links"):
-            datasets.append(dataset.get('dataset').get('external_id'))
-
-        members = {}
-        for member in data_product.get("memberships"):
-            # EXT is never present in user name?"
-            role = 'admin' if member.get('role') == 'owner' else 'member'
-            members[member.get('user').get('email')] = role
-        data_products[data_product.get("external_id")] = {"description": data_product.get("description"), "read_datasets": datasets, "users": members}
-    with open(os.path.join(FOLDER, 'config', 'data_product_glossary', 'data_product_glossary.yaml'), 'w') as f:
-        yaml.dump(data_products, f, allow_unicode=True)
+        for dataset_link in data_product.get("dataset_links"):
+            if dataset_link.get("status") == "approved":
+                datasets.append(dataset_link.get("dataset").get("external_id"))
+
+        # members = {}
```

Comment:
why do you remove all of the memberships / owners part?
We need to have this available somewhere, currently we put it in the yaml. To e.g. create Conveyor accounts for members

Reply:
I was unsure if this currently had any use, I had commented it out as it was unnecessary for the AWS integration. I added it back to the export